Controller/Customer/DeleteCreditCard.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2019 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Controller\Customer;
+use Magento\Framework\App\Action\Action;
+use Bolt\Boltpay\Model\CustomerCreditCardFactory;
+use Magento\Framework\App\Action\Context;
+use Bolt\Boltpay\Helper\Bugsnag;
+use Magento\Framework\Data\Form\FormKey\Validator;
+
+class DeleteCreditCard extends Action
+{
+    /**
+     * @var \Bolt\Boltpay\Model\CustomerCreditCard
+     */
+    private $customerCreditCardFactory;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @var Validator
+     */
+    private $formKeyValidator;
+
+    /**
+     * DeleteCreditCard constructor.
+     * @param Context $context
+     * @param Bugsnag $bugsnag
+     * @param CustomerCreditCardFactory $customerCreditCardFactory
+     * @param Validator $formKeyValidator
+     */
+    public function __construct(
+        Context $context,
+        Bugsnag $bugsnag,
+        CustomerCreditCardFactory $customerCreditCardFactory,
+        Validator $formKeyValidator
+    )
+    {
+        $this->formKeyValidator = $formKeyValidator;
+        $this->bugsnag = $bugsnag;
+        $this->customerCreditCardFactory = $customerCreditCardFactory;
+        parent::__construct($context);
+    }
+
+    /**
+     * @return \Magento\Framework\App\ResponseInterface|\Magento\Framework\Controller\ResultInterface|void
+     */
+    public function execute()
+    {
+        try {
+            if (!$this->formKeyValidator->validate($this->getRequest())) {
+                $this->messageManager->addErrorMessage(__('Invalid form key'));
+                return $this->_redirect($this->_redirect->getRefererUrl());
+            }
+
+            $id = $this->getRequest()->getParam('id');
+            if($id){
+                $creditCard = $this->customerCreditCardFactory->create()->load($id);
+
+                if($creditCard && $creditCard->getId()){
+                    $creditCard->delete();
```

Comment:
Can this op fail? What happens then?

Reply:
@matt-thomason Technically yes, but I didn’t receive any error when I was testing the code. If this ever fails, the page will be reloaded, the error message will be shown to the customer at the top of the listing credit card page and it will also be logged in Bugsnag.

Reply:
Fair enough!